index.bs

```diff
@@ -0,0 +1,250 @@
+as it is written only for handling non-[=Document/fully active=] (BFCached) documents.
+
+In many cases,
+anything that happens while the document is not [=Document/fully active=]
+should be treated as if it never happened.
+If it makes more sense to "update" a document to ensure it does not hold  stale information
+after it becomes [=Document/fully active=] again, consider the [[#listen-fully-active]] pattern below.
+<div class="example">
+  APIs that periodically send information updates,
+  such as Geolocation API's {{Geolocation/watchPosition()}}
+  should not send updates if the document is no longer fully active.
+  They also should not queue those updates to arrive later.
+  Instead, they should only resume sending updates when the document becomes active again,
+  possibly sending one update with the latest information then.
+</div>
+
+Note: If the actions are already protected by certain checks that can only be satisfied if the document is [=Document/fully active=],
+such as checking if the top-level browsing context has [=system focus=],
+fully active checks might not be needed.
+However, be careful of certain checks like transient user activation,
+which can be true even if a document is not fully active.
+See also the [[#per-document-state]] section.
+
+<h4 id="listen-fully-active">Listen for changes to [=Document/fully active=] status</h4>
+
+When a document goes from [=Document/fully active=] to non-[=Document/fully active=],
+it should be treated similarly to the way discarded documents are treated.
+The document must not retain exclusive access to shared resources
+and must ensure that no new requests are issued
+and that connections that allow for new incoming requests are terminated.
+When a document goes from non-[=Document/fully active=] to [=Document/fully active=] again,
+it can restore connections if appropriate.
+
+While web authors can manually do cleanup (e.g. release the resources, sever connections)
+from within the {{pagehide}} event and restore them from the {{pageshow}} event themselves,
+doing this automatically from the API design allows the document to be kept alive after navigation by default,
+and is more likely to lead to well-functioning web applications.
+
+<div class="example">
+  APIs that create live connections can pause/close the connection and possibly resume/reopen it later.
+  It's also possible to let the connection stay open to complete existing ongoing requests,
+  and later update the document with the result when it gets restored, if appropriate (e.g.
+  resource loads).
+</div>
+<div class="example">
+  APIs that hold non-exclusive resources
+  may be able to release the resource when the document becomes not fully active,
+  and re-acquire them when it becomes [=Document/fully active=] again
+  (Screen Wake Lock API is already <a href="https://w3c.github.io/screen-wake-lock/#handling-document-loss-of-full-activity">doing</a> the first part).
```

Comment:
This is pretty poor example in that there's no explicit hook tying this to some algorithm that runs when the fully active state changes. Are there hooks for that?

Reply:
Unfortunately we don't have hooks for "fully active" state changes. Maybe we should add that? Do you know what's the best way to do that?

Reply:
I suspect we want something similar to HTML's "update the rendering" where specifications have an algorithm that HTML ends up invoking when it changes the state.

Reply:
These are all valid points though not necessarily strong enough to hold back the forking of this document. My suggestion is to open separate issues against the newly forked BFCache document.

Reply:
I find that a bit rude. I was asked for feedback here and only now you're telling me I should have been filing issues instead.

Reply:
Hi @atanassov, I think the merge of this PR wasn't urgently needed. The HTML file of the bikeshed file is already part of the gh-pages branch so the document was already accessible from https://w3ctag.github.io/bfcache-guide/. I had intended to merge the reviewed version of the document to that branch after all the issues are resolved and this got LGTMs from other browsers.

@annevk, I'm very sorry for the delay in addressing the reviews here and the sudden merge, I assure you that I will address the issues and update the guide. I have filed [an issue](https://github.com/w3ctag/bfcache-guide/issues/2) for the "in parallel" one, and [another issue](https://github.com/whatwg/html/issues/8872) for the "fully active" state change listener. Thanks a lot for reviewing this PR and all the help!!